internal/controller/controllers/agent_controller.go

```diff
@@ -80,69 +97,208 @@ func (r *AgentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+		})
 	}
+}
+
+func (r *AgentReconciler) validated(agent *adiiov1alpha1.Agent, status string, h *models.Host) {
+	validationInfo := ""
+	validationRes, err := host.GetValidations(h)
+	var failures []string
+	if err == nil {
+		for _, vRes := range validationRes {
+			for _, v := range vRes {
+				if v.Status == host.ValidationFailure {
+					failures = append(failures, v.Message)
+				}
+			}
+		}
+		validationInfo = strings.Join(failures[:], ",")
+	}
+	if status == models.HostStatusInsufficient {
+		conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+			Type:    adiiov1alpha1.ValidatedCondition,
+			Status:  corev1.ConditionFalse,
+			Reason:  adiiov1alpha1.AgentValidationsFailingReason,
+			Message: fmt.Sprintf("%s %s", adiiov1alpha1.AgentValidationsFailingMsg, validationInfo),
+		})
+	} else if h.ValidationsInfo == "" {
+		conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+			Type:    adiiov1alpha1.ValidatedCondition,
+			Status:  corev1.ConditionUnknown,
+			Reason:  adiiov1alpha1.AgentValidationsUnknownReason,
+			Message: adiiov1alpha1.AgentValidationsUnknownMsg,
+		})
+	} else {
+		conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+			Type:    adiiov1alpha1.ValidatedCondition,
+			Status:  corev1.ConditionTrue,
+			Reason:  adiiov1alpha1.AgentValidationsPassingReason,
+			Message: adiiov1alpha1.AgentValidationsPassingMsg,
+		})
+	}
+}
+
+func (r *AgentReconciler) connected(agent *adiiov1alpha1.Agent, status string) {
+	if status == models.HostStatusDisconnected {
+		conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+			Type:    adiiov1alpha1.ConnectedCondition,
+			Status:  corev1.ConditionFalse,
+			Reason:  adiiov1alpha1.AgentDisconnectedReason,
+			Message: adiiov1alpha1.AgentDisonnectedMsg,
+		})
+	} else {
+		conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+			Type:    adiiov1alpha1.ConnectedCondition,
+			Status:  corev1.ConditionTrue,
+			Reason:  adiiov1alpha1.AgentConnectedReason,
+			Message: adiiov1alpha1.AgentConnectedMsg,
+		})
+	}
+}
+func (r *AgentReconciler) readyForInstallation(agent *adiiov1alpha1.Agent, status string) {
+	if status == models.HostStatusKnown {
+		conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+			Type:    adiiov1alpha1.ReadyForInstallationCondition,
+			Status:  corev1.ConditionTrue,
+			Reason:  adiiov1alpha1.AgentReadyReason,
+			Message: adiiov1alpha1.AgentReadyMsg,
+		})
+	} else if funk.ContainsString(notReadyForInstallStatuses, status) {
+		conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+			Type:    adiiov1alpha1.ReadyForInstallationCondition,
+			Status:  corev1.ConditionFalse,
+			Reason:  adiiov1alpha1.AgentNotReadyReason,
+			Message: adiiov1alpha1.AgentNotReadyMsg,
+		})
+	} else if funk.ContainsString(installingStatuses, status) {
```

Comment:
What would be the default if no condition will match?

Reply:
Added : `Unknown, "UnknownStatus", "The installation status is currently not recognized"`